Refactor kafka log directory setup

ryan-gang opened the pull request:
<!-- GitButler Footer Boundary Top -->
---
This is **part 2 of 3 in a stack** made with GitButler:
- <kbd>&nbsp;3&nbsp;</kbd> #88 
- <kbd>&nbsp;2&nbsp;</kbd> #84 👈 
- <kbd>&nbsp;1&nbsp;</kbd> #80 
<!-- GitButler Footer Boundary Bottom -->

rohitpaulk commented on internal/test_helpers/fixtures/produce/pass:
```diff
@@ -0,0 +1,932 @@
+[33m[tester::#CK2] [Serializer] [0m[36m  - Wrote file to: /tmp/kraft-combined-logs/foo-0/partition.metadata[0m
+[33m[tester::#CK2] [Serializer] [0m[36m  - Wrote file to: /tmp/kraft-combined-logs/pax-0/partition.metadata[0m
+[33m[tester::#CK2] [Serializer] [0m[36m  - Wrote file to: /tmp/kraft-combined-logs/pax-1/partition.metadata[0m
+[33m[tester::#CK2] [Serializer] [0m[36m  - Wrote file to: /tmp/kraft-combined-logs/paz-0/partition.metadata[0m
```
@ryan-gang we need to trim these logs down, the volume here is crazy - looks like this is only because we're creating topics/partitions for ALL stages in the same place?

ryan-gang commented on internal/test_helpers/fixtures/fetch/pass:
```diff
@@ -363,6 +363,7 @@ Debug = true
 [33m[tester::#GS0] [Decoder] [0m[36m  - .throttle_time_ms (0)[0m
 [33m[tester::#GS0] [Decoder] [0m[36m  - .TAG_BUFFER[0m
 [33m[tester::#GS0] [0m[92m✓ correlation_id: 177586623[0m
+[33m[tester::#GS0] [0m[92m✓ correlation_id: 177586623[0m
```
Fixed.

ryan-gang commented:
@rohitpaulk 
- Renamed `TopicRecord` and `PartitionRecord` to `ClusterMetadataTopicRecord` and `ClusterMetadataPartitionRecord`.
- Added `cluster_metadata_payload_builder`
- Added `cluster_metadata_partition_record_builder`
- Added `cluster_metadata_topic_record_builder`
- Refactor manually created records to use builders across entire generation stack
- Individual stages now need to mention which topics they want to generate setup logs for 
- By default only minimal files are created (4 files).

ryan-gang commented on protocol/serializer/cluster_metadata.go:
```diff
@@ -5,242 +5,49 @@ import (
-	}
+	recordBatch1 := builder.NewRecordBatchBuilder().
+		AddRecord(nil, GetEncodedBytes(featureLevelRecord), []kafkaapi.RecordHeader{}).
+		Build()
```
It should be 0 only. 
Checked what kafka writes to the log file.

rohitpaulk commented on protocol/common/constants.go:
```diff
@@ -21,48 +32,76 @@ const (
 	TOPIC3_NAME = topic_names[2]
-	random_topic_uuids := random.RandomInts(10, 100, 3)
+	TOPIC4_NAME = topic_names[3]
+	random_topic_uuids := random.RandomInts(10, 100, 4)
```
Let's avoid "pre-creating" these randomly here and stick to our usual pattern of creating random values on the fly in each stage. You can expose something like `getRandomTopicConfig()` / `getRandomTopicConfigs()` if you'd like and use that. 

From a user's perspective, let's have the stage functions read more cleanly and make it clear that values are generated randomly on the fly.
